src/inet/EndPointStateOpenThread.h

```diff
@@ -43,6 +52,21 @@ class DLL_EXPORT EndPointStateOpenThread
         UDP     = 1,
         TCP     = 2
     } mOpenThreadEndPointType;
+
+    void LockOpenThread(void)
+    {
+        if (lockOpenThread != nullptr)
+            lockOpenThread();
+    }
+    void UnlockOpenThread(void)
+    {
+        if (unlockOpenThread != nullptr)
+            unlockOpenThread();
+    }
```

Comment:
Suggest adding an inner class `OpenThreadLocker` that calls `LockOpenThread` in the constructor, and calls `UnlockOpenThread` in the destructor. This would give you an RAII locker that can be put in a given scope and would avoid any change of forgetting an UnlockOpenThread due to a new early return path

Reply:
Opened a follow up.